Fix a segfault that happens when $message_cachedir is set and a POP3 mailbox is
being opened, patch fetched from upstream bug http://bugs.mutt.org/3457

--- a/muttlib.c
+++ b/muttlib.c
@@ -1962,6 +1962,7 @@
 {
   char *p = safe_strdup (src);
   int rc = mutt_convert_string (&p, Charset, "utf-8", 0);
-  strfcpy (dest, rc == 0 ? p : src, dlen);
+  /* `src' may be NULL, such as when called from the pop3 driver. */
+  strfcpy (dest, (rc == 0) ? NONULL(p) : NONULL(src), dlen);
   FREE (&p);
 }
